sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/_tests_/assessment-workflow-queue-retry.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,187 @@
+import { Job, Queue } from "bull";
+import { createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames, addHours } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../../test/helpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  getQueueProviderName,
+  saveFakeApplication,
+} from "@sims/test-utils";
+import { Not } from "typeorm";
+import { StudentAssessmentStatus } from "@sims/sims-db";
+import {
+  CancelAssessmentQueueInDTO,
+  QueueService,
+  StartAssessmentQueueInDTO,
+} from "@sims/services/queue";
+import { WorkflowQueueRetryScheduler } from "../assessment-workflow-queue-retry.scheduler";
+
+describe(
+  describeProcessorRootTest(QueueNames.AssessmentWorkflowQueueRetry),
+  () => {
+    let app: INestApplication;
+    let db: E2EDataSources;
+    let retryProcessor: WorkflowQueueRetryScheduler;
+    let startApplicationAssessmentQueueMock: Queue<StartAssessmentQueueInDTO>;
+    let cancelApplicationAssessmentQueueMock: Queue<CancelAssessmentQueueInDTO>;
+    let queueService: QueueService;
+
+    beforeAll(async () => {
+      const { nestApplication, dataSource } = await createTestingAppModule();
+      app = nestApplication;
+      db = createE2EDataSources(dataSource);
+      // Mocked StartApplicationAssessment queue.
+      startApplicationAssessmentQueueMock = app.get(
+        getQueueProviderName(QueueNames.StartApplicationAssessment),
+      );
+      // Mocked CancelApplicationAssessment queue.
+      cancelApplicationAssessmentQueueMock = app.get(
+        getQueueProviderName(QueueNames.CancelApplicationAssessment),
+      );
+      // Processor under test.
+      retryProcessor = app.get(WorkflowQueueRetryScheduler);
+      queueService = app.get(QueueService);
+    });
+
+    beforeEach(async () => {
+      jest.resetAllMocks();
+      // Ensure that all assessments will be cancelled.
+      await db.studentAssessment.update(
+        {
+          studentAssessmentStatus: Not(StudentAssessmentStatus.Cancelled),
+        },
+        { studentAssessmentStatus: StudentAssessmentStatus.Cancelled },
+      );
+    });
+
+    it(`Should queue an assessment with ${StudentAssessmentStatus.Queued} status when it has this status for more than the time configured for the queue.`, async () => {
+      // Arrange
+      const application = await saveFakeApplication(db.dataSource);
+      const currentAssessment = application.currentAssessment;
+      currentAssessment.studentAssessmentStatus =
+        StudentAssessmentStatus.Queued;
+      const amountHoursAssessmentRetry =
```

Comment:
Is there a reason to not retrieve this once during the `beforeAll` as suggested here: https://github.com/bcgov/SIMS/pull/2511#discussion_r1397748562

Reply:
I moved that.